pyVmomi/SoapAdapter.py

```diff
@@ -1072,27 +1072,12 @@ def __init__(self, proxyPath):
    # specified by proxyPath.  If successful, a new HTTPSConnection is returned.
    #
    # @param path The destination URL path.
-   # @param key_file The SSL key file to use when wrapping the socket.
-   # @param cert_file The SSL certificate file to use when wrapping the socket.
-   # @param kwargs In case caller passed in extra parameters not handled by
-   #        SSLTunnelConnection
-   def __call__(self, path, key_file=None, cert_file=None, **kwargs):
-      # Only pass in the named arguments that HTTPConnection constructor
-      # understands
-      tmpKwargs = {}
-      for key in http_client.HTTPConnection.__init__.__code__.co_varnames:
-         if key in kwargs and key != 'self':
-            tmpKwargs[key] = kwargs[key]
-      tunnel = http_client.HTTPConnection(path, **tmpKwargs)
-      tunnel.request('CONNECT', self.proxyPath)
-      resp = tunnel.getresponse()
-      if resp.status != 200:
-        raise http_client.HTTPException("{0} {1}".format(resp.status, resp.reason))
-      retval = http_client.HTTPSConnection(path)
-      retval.sock = _SocketWrapper(tunnel.sock,
-                                   keyfile=key_file, certfile=cert_file)
-      return retval
-
+   # @param args Arguments are ignored
+   # @param kwargs Arguments for HTTPSConnection
+   def __call__(self, path, *args, **kwargs):
+      conn = http_client.HTTPSConnection(path, **kwargs)
+      conn.set_tunnel(self.proxyPath)
+      return conn
```

Comment:
Hi @goneri , I am still not convinced about SSLTunnel change. Any change to SSLTunnel needs to test thoroughly. This might break existing users. We have experienced this internally.
Please can you explain a bit about How Cert/Key validation happens with your change?
Provide more details about testing performed for this change.
I will also do test internally.
Thanks

Reply:
The proxy support is currently not working, so I'm not sure how this patch can break existing users.
I didn't implement the cert/key validation because we don't use it and I don't have good way to test it.

Reply:
cert Key Validation is required here.
 ```
retval.sock = _SocketWrapper(tunnel.sock,	
                                   keyfile=key_file, certfile=cert_file)
```
Please can you confirm if your change handles this? I was out of office for 2 days. I will also test in my test env and confirm
Thanks

Reply:
Hi @pgbidkar, do you have any update on this?

Reply:
Hi, We are testing your change, but test results doesn't look promising.  I will update the detailed findings by tomorrow EOD. 

Thanks

Reply:
Hi, 
Please can you test below testcases with your change and update the findings.
```
def test_postive_ssl_soapstubadapter_context_cert_key(self):
      namespace = '/vsanperf'
      stubAdapter = pyVmomi.SoapStubAdapter(
     '<HOSTNAME>',80,
      version='vim.version.version11',
     connectionPoolTimeout=10,
     sslProxyPath=namespace,
     certKeyFile=key_file,
     certFile=cert_file
    )
    vim.ServiceInstance("ServiceInstance", stubAdapter).RetrieveContent()
    
    def test_postive_SSLTunnel(self):
        vsanVerFile = '/sdk/vsanServiceVersions.xml'
        tmpArgs = {'key_file':key_file, 'cert_file':cert_file}
        scheme = SoapAdapter.SSLTunnelConnection('/sdkTunnel')
        connHandle = scheme(destAddr, **tmpArgs)
        connHandle.request('GET', vsanVerFile)
        res=connHandle.getresponse()
```

Copied methods from my editor, indentation might have gone wrong. please fix it.
Please get cert & key files for vCenter  pass them as argument.
Here Hostname or destAddress are VC IP/Hostname.

Please let me know if you find any difficulty in running

Thanks

Reply:
Why do you just pass a path (`/sdkTunnel`) and not the hostname of the proxy server?

`HTTPConnection()` exepects a hostname, see: https://docs.python.org/3/library/http.client.html

Reply:
What's the purpose of `sslProxyPath`?  From my understanding of https://github.com/vmware/pyvmomi/blob/master/pyVmomi/SoapAdapter.py#L1240-L1247 I initially thought it was an alternative way to write the `host:port` but it seems to be something different.

Reply:
> Why do you just pass a path (`/sdkTunnel`) and not the hostname of the proxy server?
> 
> `HTTPConnection()` exepects a hostname, see: https://docs.python.org/3/library/http.client.html

/sdkTunnel is tunnel proxy inbuilt in vCenter.  I didnt had actuall proxy server setup, hence using /sdkTunnel

Reply:
> What's the purpose of `sslProxyPath`? From my understanding of https://github.com/vmware/pyvmomi/blob/master/pyVmomi/SoapAdapter.py#L1240-L1247 I initially thought it was an alternative way to write the `host:port` but it seems to be something different.

'url' is the argument alternative way of sending hostname:port not sslProxyPath. It is proxy path to connect to server. It depends on the way vcenter server setup is done in your infra.

Reply:
Thank you @pgbidkar for your answer. That was really helpful! I just pushed an updated version of my patch. I now add a new `HTTPProxyConnection` class, this way, I don't touch the `SSLTunnelConnection` class. Both are really difference, and I think it's a better approach.

To summarize, if the user set `sslProxyPath`, we use the original method, if instead `httpProxyHost` is set, we use the new class. Both should not be used at the same time.